integration_test.go

```diff
@@ -1540,25 +1539,46 @@ func TestDeadlockConsumerIssue48(t *testing.T) {
 
 // https://github.com/streadway/amqp/issues/46
 func TestRepeatedChannelExceptionWithPublishAndMaxProcsIssue46(t *testing.T) {
-	if strings.Compare(os.Getenv("CI"), "true") == 0 {
-		t.Skip("FLAKY - https://github.com/rabbitmq/amqp091-go/issues/77")
-	}
-	conn := integrationConnection(t, "issue46")
-	if conn != nil {
-		t.Cleanup(func() { conn.Close() })
+	var conn *Connection = nil
 
-		for i := 0; i < 100; i++ {
-			ch, err := conn.Channel()
-			if err != nil {
-				t.Fatalf("expected error only on publish, got error on channel.open: %v", err)
+	t.Cleanup(func() {
+		if conn != nil {
+			conn.Close()
+		}
+	})
+
+	for i := 0; i < 100; i++ {
+		if conn == nil || conn.IsClosed() {
+			conn = integrationConnection(t, "issue46")
+			if conn == nil {
+				t.Fatal("conn is nil")
+			}
+		}
+
+		ch, err := conn.Channel()
+		if err, ok := err.(Error); ok {
+			if err.Code != 504 {
+				t.Fatalf("expected channel only exception i: %d got: %+v", i, err)
 			}
+		}
```

Comment:
Why was this changed from not allowing any errors to allow all errors except 504?
Doesn't this change the behavior of the test and prevents that it catches the original bug?

Reply:
In the case of a 504 you need a new connection.